### PR TITLE
Camera Refactor

### PR DESCRIPTION
**Pull request by DragonMoffon**
There are a few issues with the current camera system in arcade. Mainly it only supports a 2D orthographic camera with limited zoom and rotation functionality. This refactor aims to create a new system that allows users to easily make fully 3D orthographic or perspective cameras which are easy to use and create custom functionality for. 

The refactor aims to be mostly backwards-compatible with the previously implemented SImpleCamera and Camera.

The refactor includes:
A ViewData PoD that holds all of the information needed to make a view matrix and set the viewport. 
Two ProjectionData PoDs that hold the data for making either an orthographic or projection matrix.

The Projection Protocol to link the ProjectionData PoDs together (may be removed as it is unnecessary)
The Projector Protocol defines any object that provides the Use, Activate, and get_map_coordinates (may be renamed) methods.
The Camera Protocol defines an object which holds both a ViewData PoD and a ProjectionProtocol Object.

The OrthographicCamera class uses the ViewData and OrthographicProjectionData PoDs to provide an easy way to create and use an orthographic projection matrix, but it alone does not provide ways of controlling the camera.
The PersepctiveCamera class is the same as the OrthographicCamera class, except it uses the PerspectiveProjectionData PoD.

It provides a backwards-compatible SimpleCamera class implementation with the same methods.
The Camera2D class provides a simple interface to control nearly every camera aspect and will be what 99% of arcade users will need. It does not provide animations or any other similar logic but provides easy properties and methods for setting the different variables that are used by the camera. The only clamps on control are lack of access to the forward vector and only accepting 2D vectors.

The DefaultProjector class is provided to have a base camera only to be used internally by Arcade. This class is designed to be used as the default "current_camera". This is because of a significant bug currently plaguing Arcade. If the user has never used a Camera and they try to use a camera as a context manager, the current camera will be equal to None and cause a crash.  

See the 
#Cameras
topics in the arcade-dev forum on the Python Arcade discord server for more info.

There are also likely issues that are relevant such as #1651.

### CODE REVIEW

**Comment by DragonMoffon** [comment]
Adding to my TODO list.

Splitting the code into multiple files
Unit tests for every core piece of the code.
Examples for every camera type and a few examples of controllers.

File: arcade/cinematic/default.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+from pyglet.math import Mat4
+
+from arcade.cinematic.types import Projector
+from arcade.window_commands import get_window
+if TYPE_CHECKING:
+    from arcade.application import Window
+
+
+class DefaultProjector:
```
Comment (gran4) [inline arcade/cinematic/default.py]: Make it _DefaultProjecter? if you do not want it to be used by a developer.

Comment (DragonMoffon) [reply arcade/cinematic/default.py]: Good idea, thank you!

**Comment by DragonMoffon** [comment]
Moving branch to better reflect the goals of this PR

**Pull request closed without merging**